bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java

```diff
@@ -287,7 +301,9 @@ private void sendData(String key, String value) {
             if (logger.isDebugEnabled()) {
                 logger.debug("Response: {}", contentResponse.getContentAsString());
             }
-            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
+            GoEStatusResponseDTO result = gson.fromJson(contentResponse.getContentAsString(),
+                    GoEStatusResponseDTO.class);
+            updateChannelsAndStatus(result);
         } catch (InterruptedException | TimeoutException | ExecutionException e) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
                     "No response received on command");
```

Comment:
I would make sure to either log the exception or include the exception message as part of the status details. 
There is a big difference between a device not responding and the device returning erroneous data. It is beneficial for the user to know which one it is.
Also make sure to catch JsonSyntaxException!
This also applies to the`getGoEData()` below.

Reply:
So how detailedcan the messages of the thing status update be? I guess a complete exception stacktrace is too much, but the message itself should be ok?

Reply:
Using just the message is perfectly fine.

Reply:
So i guess i can just use 
```
catch (Exception e) {
  updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
                    e.getMessage());
}
```

That also includes json exceptions then

Reply:
Nope. Never catch `Exception`. But you can catch several exceptions in one catch-clause, e.g. 

```
try {
    ....
} catch (IOException | JsonException | WhateverException e) {
    do your exception handling here
}
```